### PR TITLE
Update to a single repo structure

### PR DESCRIPTION
**Pull request by lemonmade**
This PR gets rid of the submodule junk and adds some required tooling for doing everything in a single repo. I totally mangled the 6.x versions with my experimentation (whoops), so after this ships, I will update to 7.0.0 which will be the new version going forward. This also updates the rules that we changed our minds on from the styleguide PR: `prefer-const` switches from `0` to `1`, and `init-declarations` switches from `1` to `0`.

cc/ @bouk sorry for the huge mess :P

### CODE REVIEW

**Comment by bouk** [comment]
Could you add a script/setup to run the required set up commands?

On Friday, 5 February 2016, Chris Sauvé <notifications@github.com> wrote:

> This PR gets rid of the submodule junk and adds some required tooling for
> doing everything in a single repo. I totally mangled the 6.x versions with
> my experimentation (whoops), so after this ships, I will update to 7.0.0
> which will be the new version going forward. This also updates the rules
> that we changed our minds on from the styleguide PR: prefer-const
> switches from 0 to 1, and init-declarations switches from 1 to 0.
>
> cc/ @bouk <https://github.com/bouk> sorry for the huge mess :P
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Shopify/javascript/pull/3
> Commit Summary
>
>    - Update to a single repo structure
>
> File Changes
>
>    - *A* .eslintignore
>    <https://github.com/Shopify/javascript/pull/3/files#diff-0> (3)
>    - *A* .gitignore
>    <https://github.com/Shopify/javascript/pull/3/files#diff-1> (8)
>    - *D* .gitmodules
>    <https://github.com/Shopify/javascript/pull/3/files#diff-2> (12)
>    - *A* .nvmrc
>    <https://github.com/Shopify/javascript/pull/3/files#diff-3> (1)
>    - *A* CHANGELOG.md
>    <https://github.com/Shopify/javascript/pull/3/files#diff-4> (3)
>    - *A* CONTRIBUTING.md
>    <https://github.com/Shopify/javascript/pull/3/files#diff-5> (25)
>    - *M* README.md
>    <https://github.com/Shopify/javascript/pull/3/files#diff-6> (9)
>    - *A* VERSION
>    <https://github.com/Shopify/javascript/pull/3/files#diff-7> (1)
>    - *A* circle.yml
>    <https://github.com/Shopify/javascript/pull/3/files#diff-8> (9)
>    - *A* package.json
>    <https://github.com/Shopify/javascript/pull/3/files#diff-9> (27)
>    - *D* packages/eslint-config-shopify
>    <https://github.com/Shopify/javascript/pull/3/files#diff-10> (1)
>    - *A* packages/eslint-config-shopify/.editorconfig
>    <https://github.com/Shopify/javascript/pull/3/files#diff-11> (10)
>    - *A* packages/eslint-config-shopify/.eslintignore
>    <https://github.com/Shopify/javascript/pull/3/files#diff-12> (1)
>    - *A* packages/eslint-config-shopify/.eslintrc
>    <https://github.com/Shopify/javascript/pull/3/files#diff-13> (7)
>    - *A* packages/eslint-config-shopify/.npmignore
>    <https://github.com/Shopify/javascript/pull/3/files#diff-14> (3)
>    - *A* packages/eslint-config-shopify/LICENSE
>    <https://github.com/Shopify/javascript/pull/3/files#diff-15> (21)
>    - *A* packages/eslint-config-shopify/README.md
>    <https://github.com/Shopify/javascript/pull/3/files#diff-16> (63)
>    - *A* packages/eslint-config-shopify/core/index.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-17> (17)
>    - *A* packages/eslint-config-shopify/es5/index.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-18> (14)
>    - *A* packages/eslint-config-shopify/es6/index.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-19> (20)
>    - *A* packages/eslint-config-shopify/package.json
>    <https://github.com/Shopify/javascript/pull/3/files#diff-20> (33)
>    - *A* packages/eslint-config-shopify/react/index.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-21> (20)
>    - *A* packages/eslint-config-shopify/rules/best-practices.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-22> (121)
>    - *A* packages/eslint-config-shopify/rules/ecmascript-6.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-23> (38)
>    - *A* packages/eslint-config-shopify/rules/legacy.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-24> (16)
>    - *A* packages/eslint-config-shopify/rules/node.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-25> (22)
>    - *A* packages/eslint-config-shopify/rules/possible-errors.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-26> (60)
>    - *A* packages/eslint-config-shopify/rules/react.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-27> (84)
>    - *A* packages/eslint-config-shopify/rules/shopify.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-28> (6)
>    - *A* packages/eslint-config-shopify/rules/strict-mode.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-29> (6)
>    - *A* packages/eslint-config-shopify/rules/stylistic-issues.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-30> (112)
>    - *A* packages/eslint-config-shopify/rules/variables.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-31> (26)
>    - *D* packages/eslint-plugin-shopify
>    <https://github.com/Shopify/javascript/pull/3/files#diff-32> (1)
>    - *A* packages/eslint-plugin-shopify/.babelrc
>    <https://github.com/Shopify/javascript/pull/3/files#diff-33> (3)
>    - *A* packages/eslint-plugin-shopify/.editorconfig
>    <https://github.com/Shopify/javascript/pull/3/files#diff-34> (10)
>    - *A* packages/eslint-plugin-shopify/.eslintignore
>    <https://github.com/Shopify/javascript/pull/3/files#diff-35> (2)
>    - *A* packages/eslint-plugin-shopify/.eslintrc
>    <https://github.com/Shopify/javascript/pull/3/files#diff-36> (3)
>    - *A* packages/eslint-plugin-shopify/.npmignore
>    <https://github.com/Shopify/javascript/pull/3/files#diff-37> (6)
>    - *A* packages/eslint-plugin-shopify/LICENSE
>    <https://github.com/Shopify/javascript/pull/3/files#diff-38> (21)
>    - *A* packages/eslint-plugin-shopify/README.md
>    <https://github.com/Shopify/javascript/pull/3/files#diff-39> (58)
>    - *A*
>    packages/eslint-plugin-shopify/docs/rules/binary-assignment-parens.md
>    <https://github.com/Shopify/javascript/pull/3/files#diff-40> (33)
>    - *A* packages/eslint-plugin-shopify/docs/rules/require-flow.md
>    <https://github.com/Shopify/javascript/pull/3/files#diff-41> (35)
>    - *A* packages/eslint-plugin-shopify/index.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-42> (11)
>    - *A*
>    packages/eslint-plugin-shopify/lib/rules/binary-assignment-parens.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-43> (55)
>    - *A* packages/eslint-plugin-shopify/lib/rules/require-flow.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-44> (44)
>    - *A* packages/eslint-plugin-shopify/package.json
>    <https://github.com/Shopify/javascript/pull/3/files#diff-45> (34)
>    - *A*
>    packages/eslint-plugin-shopify/tests/lib/rules/binary-assignment-parens.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-46> (156)
>    - *A* packages/eslint-plugin-shopify/tests/lib/rules/require-flow.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-47> (68)
>    - *D* packages/generator-eslint-shopify
>    <https://github.com/Shopify/javascript/pull/3/files#diff-48> (1)
>    - *A* packages/generator-eslint-shopify/.babelrc
>    <https://github.com/Shopify/javascript/pull/3/files#diff-49> (7)
>    - *A* packages/generator-eslint-shopify/.editorconfig
>    <https://github.com/Shopify/javascript/pull/3/files#diff-50> (11)
>    - *A* packages/generator-eslint-shopify/.eslintignore
>    <https://github.com/Shopify/javascript/pull/3/files#diff-51> (3)
>    - *A* packages/generator-eslint-shopify/.eslintrc
>    <https://github.com/Shopify/javascript/pull/3/files#diff-52> (14)
>    - *A* packages/generator-eslint-shopify/.npmignore
>    <https://github.com/Shopify/javascript/pull/3/files#diff-53> (6)
>    - *A* packages/generator-eslint-shopify/LICENSE
>    <https://github.com/Shopify/javascript/pull/3/files#diff-54> (21)
>    - *A* packages/generator-eslint-shopify/README.md
>    <https://github.com/Shopify/javascript/pull/3/files#diff-55> (49)
>    - *A* packages/generator-eslint-shopify/package.json
>    <https://github.com/Shopify/javascript/pull/3/files#diff-56> (61)
>    - *A* packages/generator-eslint-shopify/src/app/index.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-57> (87)
>    - *A* packages/generator-eslint-shopify/test/.eslintrc
>    <https://github.com/Shopify/javascript/pull/3/files#diff-58> (15)
>    - *A* packages/generator-eslint-shopify/test/helper.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-59> (8)
>    - *A* packages/generator-eslint-shopify/test/src/app/index.test.js
>    <https://github.com/Shopify/javascript/pull/3/files#diff-60> (64)
>    - *D* packages/shopify-flow-fixes
>    <https://github.com/Shopify/javascript/pull/3/files#diff-61> (1)
>
> Patch Links:
>
>    - https://github.com/Shopify/javascript/pull/3.patch
>    - https://github.com/Shopify/javascript/pull/3.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/javascript/pull/3>.
>


-- 
Bouke

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+machine:
+  node:
+    version: 5.5.0
+
+test:
+  pre:
+    - script/setup
```
Comment (bouk) [inline circle.yml]: This should be in `dependencies/override`

File: packages/eslint-config-shopify/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2015 Brigade
```
Comment (bouk) [inline packages/eslint-config-shopify/LICENSE]: `Brigade`?

File: packages/eslint-config-shopify/core/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+var merge = require('merge');
+
+module.exports = {
+  plugins: [
+    'shopify',
+  ],
+
+  rules: merge(
```
Comment (bouk) [inline packages/eslint-config-shopify/core/index.js]: Won't `Object.assign` do in this case? It's not a deep merge

File: packages/eslint-plugin-shopify/lib/rules/binary-assignment-parens.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+/* eslint no-unused-vars:0, no-console: 0 */
```
Comment (bouk) [inline packages/eslint-plugin-shopify/lib/rules/binary-assignment-parens.js]: Why this? There's no console in this file

File: packages/generator-eslint-shopify/LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2016 Chris Sauve <chris.sauve@shopify.com>
```
Comment (bouk) [inline packages/generator-eslint-shopify/LICENSE]: `Chris Sauve` -> `Shopify`

**Comment by bouk** [comment]
some small things but lgtm

File: circle.yml (hunk 1 above)
Comment (lemonmade) [inline circle.yml]: Thanks!

File: packages/eslint-config-shopify/core/index.js (hunk 3 above)
Comment (lemonmade) [inline packages/eslint-config-shopify/core/index.js]: :+1:

**Pull request merged**